src/index.ts

```diff
@@ -51,23 +64,38 @@ export const getMatchingFilePaths = async ({
       extensions.some((ext) => filename.endsWith(`.${ext}`))
   }
   if (!fileMatchFn) fileMatchFn = () => true
-  return (await glob("**/*", { cwd: dirPath, nodir: true })).filter(
-    (filename) => fileMatchFn!(filename, path.resolve(dirPath, filename))
-  )
+
+  const files = await glob("**/*", { 
+    cwd: dirPath,
+    nodir: true,
+    dot: true,
+    absolute: false,
+    windowsPathsNoEscape: true
+  })
+  
+  return files
+    .map(f => toPosixPath(f))
+    .filter(filename => fileMatchFn!(filename, path.join(dirPath, filename)))
 }
 
 export const getVirtualFileSystemFromDirPath = async (
   opts: SearchOpts
 ): Promise<Record<Path, Content>> => {
-  const { dirPath, contentFormat } = opts
+  const { dirPath, contentFormat = "string" } = opts
   const filePaths = await getMatchingFilePaths(opts)
   const vfs: Record<Path, Content> = {}
+  
   for (const filePath of filePaths) {
-    vfs[filePath] = await fs.readFile(path.resolve(dirPath, filePath))
-    if (contentFormat === "string") {
-      vfs[filePath] = vfs[filePath].toString()
+    try {
+      const fullPath = path.join(dirPath, filePath)
+      const content = await fs.readFile(fullPath)
+      const normalizedPath = toPosixPath(filePath)
+      vfs[normalizedPath] = content.toString()
+    } catch (err) {
+      console.error(`Failed to read file ${filePath}:`, err)
```

Comment:
don't silent error! remove this

Reply:
i guess it's not a silent error but it will still be ignored and cause issues. Try improving the error message etc. but don't just ignore errors unless the user specifies to do so